### DISCUSSION TITLE
How do you enable profiling?

### DISCUSSION THREAD

**Discussion by iand675** [srid/haskell-flake#155] [Q&A]
Hi, I am trying to figure out how to enable profiling on a package I maintain (along with its dependencies). Can you provide any guidance about how to make that work with `haskell-flake`?

**Comment by srid** [comment]
In [`overrides`](https://haskell.flake.page/dependency), do:

```nix
mypackage = pkgs.haskell.lib.enableLiraryProfiling super.mypackage;
```

eg: https://github.com/srid/emanote/blob/cedd7d8cde17e0dbe96b2b9d2210d8da7d2e2d01/flake.nix#L74-L76

**Reply by iand675** [reply]
When I do that, I get a lot of errors about how the underlying libraries aren't installed with profiling support:

```
    Could not find module ‘Control.Exception’
    Perhaps you haven't installed the profiling libraries for package ‘base-4.17.0.0’?
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
```

Do I have to override all of the transitive dependencies the way you specified to make it work?

**Reply by srid** [reply]
I believe profiling is enabled by default in `pkgs.haskellPackages`. What is your `basePackages`? Your `flake.nix`?

**Reply by iand675** [reply]
Here is the flake in question. At the moment I'm trying to use the default repl on darwin `aarch64` https://github.com/iand675/hs-opentelemetry/blob/improved-dux/flake.nix

**Reply by srid** [reply]
You are using `pkgs.haskell.packages.${compilerVersion}`. I know that library profiling is enabled by default in `pkgs.haskellPackages`. You could try asking in https://app.element.io/#/room/#haskell:nixos.org

But yes if library profiling is disabled by default in those package sets, you'd have to `mapAttrs` over them and enable it in the overrides, which would trigger compilation of the whole package set (in use).
